cmd/scollector/collectors/apache_mod_info.go

```diff
@@ -0,0 +1,112 @@
+package collectors
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"golang.org/x/net/html"
+
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+)
+
+const (
+	apacheModInfoURL = "http://127.0.0.1/server-info"
+)
+
+func init() {
+	collectors = append(collectors, &IntervalCollector{F: c_apache_mod_info, Enable: enableURL(apacheModInfoURL), Interval: time.Minute * 30})
+}
+
+/* extractTimeouts processes the "?server" output and extracts timeout settings.
+More specifically it finds the line that looks like:
+ "Timeouts: connection: 60 keep-alive: 15"
+and returns the two values contained therein.
+*/
+func extractTimeouts(doc *html.Node) (int, int, error) {
+
+	// Walk the node tree.
+	// If the node is a "dt",
+	//   walk its children until we find contents "Timeouts:"
+	//   At which point we walk those children to find "connection".
+	// Else, keep walking.
+	//   Then continue walking Children to find contents starts with "connection":
+	//     Return contents
+
+	var search_children func(*html.Node) string
+	search_children = func(n *html.Node) string {
+		if n.Type == html.TextNode && strings.HasPrefix(n.Data, "connection:") {
+			return n.Data
+		}
+		for c := n.FirstChild; c != nil; c = c.NextSibling {
+			if value := search_children(c); value != "" {
+				return value
+			}
+		}
+		return ""
+	}
+
+	var walk_subtree func(*html.Node) string
+	walk_subtree = func(n *html.Node) string {
+		if n.Type == html.ElementNode && n.Data == "dt" {
+			if val := search_children(n); val != "" {
+				return val
+			}
+		} else {
+			for c := n.FirstChild; c != nil; c = c.NextSibling {
+				if val := walk_subtree(c); val != "" {
+					return val
+				}
+			}
+		}
+		return ""
+	}
+
+	original := walk_subtree(doc)
+	parts := strings.Fields(original)
+	if (len(parts) < 4 || parts[0] != "connection:") || (parts[2] != "keep-alive:") {
```

Comment:
the len(parts) < 4 check should be on its own, because we use parts[3] down on line 79.